hack/godep-save.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+pin-godep() {
+  pushd "${GOPATH}/src/github.com/tools/godep" > /dev/null
+    git checkout "$1"
+    "${GODEP}" go install
+  popd > /dev/null
+}
+
+ORIGIN_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+# build the godep tool
+# Again go get stinks, hence || true
+go get -u github.com/tools/godep 2>/dev/null || true
+GODEP="${GOPATH}/bin/godep"
+
+# Use to following if we ever need to pin godep to a specific version again
+pin-godep 'v63'
+
+# Some things we want in godeps aren't code dependencies, so ./...
+# won't pick them up.
+REQUIRED_BINS=(
+  "github.com/elazarl/goproxy"
+  "github.com/golang/mock/gomock"
+  "./..."
+)
+
+pushd "${ORIGIN_ROOT}" > /dev/null
+"${GODEP}" save "${REQUIRED_BINS[@]}"
```

Comment:
yes, -t. and i hadn't considered cherry-picks to the dep tree.  however, i'm pretty sure that ```godep save``` doesn't overwrite files that are alredy in the dep tree.  ```godep update``` is required for that. @eparis wdyt?